apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -1542,6 +1567,224 @@ class LivechatClass {
+			remove?: Pick<ILivechatDepartmentAgents, 'agentId'>[];
+		},
+	) {
+		check(_id, String);
+		check(departmentAgents, {
+			upsert: Match.Maybe([
+				Match.ObjectIncluding({
+					agentId: String,
+					username: String,
+					count: Match.Maybe(Match.Integer),
+					order: Match.Maybe(Match.Integer),
+				}),
+			]),
+			remove: Match.Maybe([
+				Match.ObjectIncluding({
+					agentId: String,
+					username: Match.Maybe(String),
+					count: Match.Maybe(Match.Integer),
+					order: Match.Maybe(Match.Integer),
+				}),
+			]),
+		});
+
+		const department = await LivechatDepartment.findOneById<Pick<ILivechatDepartment, 'enabled'>>(_id, { projection: { enabled: 1 } });
+		if (!department) {
+			throw new Meteor.Error('error-department-not-found', 'Department not found');
+		}
+
+		return updateDepartmentAgents(_id, departmentAgents, department.enabled);
+	}
+
+	async saveRoomInfo(
+		roomData: {
+			_id: string;
+			topic?: string;
+			tags?: string[];
+			livechatData?: { [k: string]: string };
+			// For priority and SLA, if the value is blank (ie ""), then system will remove the priority or SLA from the room
+			priorityId?: string;
+			slaId?: string;
+		},
+		guestData?: {
+			_id: string;
+			name?: string;
+			email?: string;
+			phone?: string;
+			livechatData?: { [k: string]: string };
+		},
+		userId?: string,
+	) {
+		this.logger.debug(`Saving room information on room ${roomData._id}`);
+		const { livechatData = {} } = roomData;
+		const customFields: Record<string, string> = {};
+
+		if ((!userId || (await hasPermissionAsync(userId, 'edit-livechat-room-customfields'))) && Object.keys(livechatData).length) {
+			const fields = LivechatCustomField.findByScope('room');
+			for await (const field of fields) {
+				if (!livechatData.hasOwnProperty(field._id)) {
+					continue;
+				}
+				const value = trim(livechatData[field._id]);
+				if (value !== '' && field.regexp !== undefined && field.regexp !== '') {
+					const regexp = new RegExp(field.regexp);
+					if (!regexp.test(value)) {
+						throw new Meteor.Error(i18n.t('error-invalid-custom-field-value', { field: field.label }));
+					}
+				}
+				customFields[field._id] = value;
+			}
+			roomData.livechatData = customFields;
+			Livechat.logger.debug(`About to update ${Object.keys(customFields).length} custom fields on room ${roomData._id}`);
+		}
+
+		await LivechatRooms.saveRoomById(roomData);
+
+		setImmediate(() => {
+			void Apps.triggerEvent(AppEvents.IPostLivechatRoomSaved, roomData._id);
+		});
+
+		if (guestData?.name?.trim().length) {
+			const { _id: rid } = roomData;
+			const { name } = guestData;
+			return (
+				(await Rooms.setFnameById(rid, name)) &&
+				(await LivechatInquiry.setNameByRoomId(rid, name)) &&
+				// This one needs to be the last since the agent may not have the subscription
+				// when the conversation is in the queue, then the result will be 0(zero)
+				Subscriptions.updateDisplayNameByRoomId(rid, name)
```

Comment:
this also returns a `Promise`, no need to `await`? what is supposed to be returned here? `true` if all actions succeed?

Reply:
IIRC, we don't care about the return value for this call, as long as it doesn't fail. 👀

Reply:
so does it makes sense to call 3 actions one by one and return a boolean? like

```ts
await Rooms.setFnameById(rid, name);
await LivechatInquiry.setNameByRoomId(rid, name);

// This one needs to be the last since the agent may not have the subscription
// when the conversation is in the queue, then the result will be 0(zero)
await Subscriptions.updateDisplayNameByRoomId(rid, name);

return true;
```
they could even all be in a `Promise.all` right?

Reply:
Think we can move the first 2 to a promise.all, and leave the last one out, as the idea of it was to have it always as the last step :eyes:

That would work.

Reply:
Actually, the promise.all worked equally well so, changed

Reply:
great! I saw the comment but didn't actually got it why, since the commands were not really related, unless there was another side effect somewhere else.. well, good that you tested it =)

Reply:
Think this has been like this for a while, since fibers. Apparently, it was like a "sequential" chain that should return true if everything worked, and the last action was left "at last" since it may return 0, so it wont work with the `&&` that were using there. Then, this was moved to the "no fibers" world in the same fashion lol